pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -142,24 +153,38 @@ public String getActualTableName(String caseInsensitiveTableName) {
    * Returns the table config for the given table, or {@code null} if it does not exist.
    */
   @Nullable
+  @Override
   public TableConfig getTableConfig(String tableNameWithType) {
     return _tableConfigMap.get(tableNameWithType);
   }
 
+  @Override
+  public List<TableConfig> registerTableConfigChangeListener(TableConfigChangeListener tableConfigChangeListener) {
```

Comment:
should be synchronized so that we don't collide with the notification thread

Reply:
+1

Reply:
i did not follow how will adding `synchronized` here will help? or do you mean add it here and also to `notifyTableConfigChangeListeners` method?

Reply:
added synchronized